docs/resources/Guild.md

```diff
@@ -945,10 +951,53 @@ Returns a PNG image widget for the guild. Requires no permissions or authenticat
 | banner2 | smaller widget style with guild icon, name and online count. Split on the right with Discord logo                                                              | [Example](https://discord.com/api/guilds/81384788765712384/widget.png?style=banner2) |
 | banner3 | large image with guild icon, name and online count. In the footer, Discord logo on the left and "Chat Now" on the right                                        | [Example](https://discord.com/api/guilds/81384788765712384/widget.png?style=banner3) |
 | banner4 | large Discord logo at the top of the widget. Guild icon, name and online count in the middle portion of the widget and a "JOIN MY SERVER" button at the bottom | [Example](https://discord.com/api/guilds/81384788765712384/widget.png?style=banner4) |
+
+## Update Self Voice State % PATCH /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/voice-states/@me
+
+Updates the current user's voice state.
+
+###### JSON Params
+
+| Field                       | Type               | Description                                    |
+| --------------------------- | ------------------ | ---------------------------------------------- |
+| channel_id                  | snowflake          | the id of the channel the user is currently in |
+| suppress?                   | boolean            | toggles the user's suppress state              |
+| request_to_speak_timestamp? | ?ISO8601 timestamp | sets the user's request to speak               |
+
+###### Caveats
+
+There are currently several caveats for this endpoint:
+
+- `channel_id` must currently point to a stage channel.
+- current user must already have joined `channel_id`.
+- You must have the `MUTE_MEMBERS` permission to unsuppress yourself. You can always suppress yourself.
+- You must have the `REQUEST_TO_SPEAK` permission to request to speak. You can always clear your own request to speak.
+- You are able to set `request_to_speak_timestamp` to any present or future time.
+
+## Update Others Voice State % PATCH /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/voice-states/{user.id#DOCS_RESOURCES_USER/user-object}
+
+Updates another user's voice state.
+
+###### JSON Params
+
+| Field      | Type      | Description                                    |
+| ---------- | --------- | ---------------------------------------------- |
+| channel_id | snowflake | the id of the channel the user is currently in |
+| suppress?  | boolean   | toggles the user's suppress state              |
+
+###### Caveats
+
+There are currently several caveats for this endpoint:
+
+- `channel_id` must currently point to a stage channel.
+- User must already have joined `channel_id`.
+- You must have the `MUTE_MEMBERS` permission. (Since suppression is the only thing that is available currently.)
+- When unsuppressed, non-bot users will have their `request_to_speak_timestamp` set to the current time. Bot users will not.
```

Comment:
We chose to not set `request_to_speak_timestamp`, so that bots would effectively "auto speak" when they were invited. The advantage is that most bots (once they are able to join the stage channel), wouldn't have to do anything to support going "onto the stage".

The disadvantage is that _maybe_ there is some unique usecase where a bot would like to accept when they got voluntold up to the stage. But we haven't been able to come up with anything substainally useful. Please note that bots are still able to "raise their hand" regardless of this behaviour.